cdap-ui/app/cdap/components/PluginSchemaEditor/index.tsx

```diff
@@ -441,7 +441,12 @@ class PluginSchemaEditorBase extends React.PureComponent<
           disabled={this.props.disabled}
           onChange={({ avroSchema }) => {
             const newSchemas = [...this.props.schemas];
-            newSchemas[i] = avroSchema;
+            const fields = objectQuery(avroSchema, 'schema', 'fields');
+            if (!Array.isArray(fields) || !fields.length) {
```

Comment:
I believe `if (!Array.isArray(fields) || !fields)` could also work.

Reply:
This will not check for empty arrays